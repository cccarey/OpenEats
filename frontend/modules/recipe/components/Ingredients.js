import React from 'react'

export default React.createClass({
  getInitialState: function() {
    return {
      data: this.props.data || []
    };
  },

  componentWillReceiveProps(nextProps) {
    this.setState({data: nextProps.data});
  },

  render: function() {
    var ingredients = this.state.data.map(function(ingredient) {
      return (
        <li className="ingredient" key={ ingredient.id }>
          { ingredient.quantity }&nbsp;
          { ingredient.measurement }&nbsp;
          { ingredient.title }
        </li>
      );
    });
    return (
      <ul className="ingredients" >
        { ingredients }
      </ul>
    );
  }
});
