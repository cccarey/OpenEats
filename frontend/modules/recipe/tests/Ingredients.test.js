import React from 'react';
import Ingredients from '../components/Ingredients';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const data = [{"id":1,"title":"black pepper","quantity":1,"measurement":"dash","recipe":1},{"id":2,"title":"chili powder","quantity":4,"measurement":"tablespoons","recipe":1},{"id":3,"title":"cumin","quantity":1,"measurement":"tablespoon","recipe":1},{"id":4,"title":"dark kidney beans","quantity":1,"measurement":"can","recipe":1},{"id":5,"title":"diced tomatos","quantity":2,"measurement":"cans","recipe":1},{"id":6,"title":"green bell pepper","quantity":1,"measurement":"whole","recipe":1},{"id":7,"title":"ground pork","quantity":1,"measurement":"pound","recipe":1},{"id":8,"title":"ground sirloin","quantity":1,"measurement":"pound","recipe":1},{"id":9,"title":"kosher salt","quantity":1,"measurement":"dash","recipe":1},{"id":10,"title":"light kidney beans","quantity":1,"measurement":"can","recipe":1},{"id":11,"title":"serrano pepper ","quantity":1,"measurement":"whole","recipe":1},{"id":12,"title":"white onion","quantity":1,"measurement":"whole","recipe":1}];
  const component = renderer.create(
    <Ingredients data={ data }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  //
  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});