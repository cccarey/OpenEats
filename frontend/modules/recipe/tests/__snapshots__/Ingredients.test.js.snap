// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<ul
  className="ingredients"
>
  <li
    className="ingredient"
  >
    1
     
    dash
     
    black pepper
  </li>
  <li
    className="ingredient"
  >
    4
     
    tablespoons
     
    chili powder
  </li>
  <li
    className="ingredient"
  >
    1
     
    tablespoon
     
    cumin
  </li>
  <li
    className="ingredient"
  >
    1
     
    can
     
    dark kidney beans
  </li>
  <li
    className="ingredient"
  >
    2
     
    cans
     
    diced tomatos
  </li>
  <li
    className="ingredient"
  >
    1
     
    whole
     
    green bell pepper
  </li>
  <li
    className="ingredient"
  >
    1
     
    pound
     
    ground pork
  </li>
  <li
    className="ingredient"
  >
    1
     
    pound
     
    ground sirloin
  </li>
  <li
    className="ingredient"
  >
    1
     
    dash
     
    kosher salt
  </li>
  <li
    className="ingredient"
  >
    1
     
    can
     
    light kidney beans
  </li>
  <li
    className="ingredient"
  >
    1
     
    whole
     
    serrano pepper 
  </li>
  <li
    className="ingredient"
  >
    1
     
    whole
     
    white onion
  </li>
</ul>
`;
